pyvrp/cpp/TimeWindowSegment.h

```diff
@@ -131,10 +131,13 @@ TimeWindowSegment::merge(Matrix<Duration> const &durationMatrix,
     Dur const edgeDuration = durationMatrix(idxLast_, other.idxFirst_);
     Dur const atOther = duration_ - timeWarp_ + edgeDuration;
 
-    // Wait duration increases if we arrive at the other's first client before
-    // opening, whereas time warp increases if we arrive there after closing.
-    Dur const diffWait = std::max<Dur>(other.twEarly_ - atOther - twLate_, 0);
+    // Time warp increases if we arrive at the other's first client after its
+    // time window closes, whereas wait duration increases if we arrive there
+    // before opening.
     Dur const diffTw = std::max<Dur>(twEarly_ + atOther - other.twLate_, 0);
```

Comment:
Question: why no ternary for `diffTw`? Here's also a subtraction by `other.twLate` that could cause underflow.

Reply:
Related: in line 174 we also have an expression which subtract `twLate`:
```
         + std::max<Duration>(releaseTime_ - twLate_, 0)
```

Reply:
We have:
- `twEarly_ >= 0`
- `atOther >= 0`
- `releaseTime_ >= 0`

And then a subtraction by everything up to and including `INT_MAX` is well-defined. So these cases are not problematic.